### DISCUSSION TITLE
How to use my own date type with pgx

### DISCUSSION THREAD

**Discussion by f9a** [jackc/pgx#1742] [Q&A] [answered]
👋 

I would like to save the value of a column (TEXT) into a struct defined by me.
For this I have implemented the faster BinaryDecoder interface as mentioned in https://github.com/jackc/pgx/issues/1132#issuecomment-1001645249. However, the method is not called when Scan is executed. I use pgxpool to connect to the database. Is somewhere described when the interfaces 
pgtype.Value, pgtype.BinaryDecoder, pgtype.TextDecoder, pgtype.BinaryEncoder, and pgtype.TextEncoder are used? 

Thanks for your help and this nice package 🙏

**Comment by jackc** [comment, marked as answer]
pgtype.BinaryDecoder, pgtype.TextDecoder, pgtype.BinaryEncoder, and pgtype.TextEncoder are part of pgx v4. They are not part of v5. That's why they're not being called.

You probably want to implement `pgtype.TextScanner`.
